addons/godot-firebase/storage/storage.gd

```diff
@@ -271,11 +271,12 @@ func _finish_request(result : int) -> void:
             task.data = JSON.parse(_response_data.get_string_from_utf8()).result
     
     task.finished = true
-    if task.data.has("error"):
-        emit_signal("task_failed", task.result, task.response_code, task.data)
-    else:
-        emit_signal("task_successful", task.result, task.response_code, task.data)
     task.emit_signal("task_finished")
+    if typeof(task.data) == TYPE_DICTIONARY:
```

Comment:
Better, but that return statement prevents crucial code from executing at the bottom of be function. Use an if else statement like you had it before.

Reply:
oops, didn't notice that portion of code below. It was folded :P

Reply:
If I use an else statement, the `task_successful` signal would be emitted only if task.data is a dictionary

Reply:
How about this?
```gdscript
if typeof(task.data) == TYPE_DICTIONARY and task.data.has("error"):
    emit_signal("task_failed", task.result, task.response_code, task.data)
else:
    emit_signal("task_successful", task.result, task.response_code, task.data)
```

Reply:
You are right, I thought that the secound statement could be evaluated anyway and give an error, but that happens only if the type is inferred.